src/qt/choice.cpp

```diff
@@ -10,38 +10,76 @@
 
 #include "wx/choice.h"
 #include "wx/qt/private/winevent.h"
+
 #include <QtWidgets/QComboBox>
+#include <QtCore/QSortFilterProxyModel>
 
-class wxQtChoice : public wxQtEventSignalHandler< QComboBox, wxChoice >
+namespace
 {
-public:
-    wxQtChoice( wxWindow *parent, wxChoice *handler );
+    class LexicalSortProxyModel : public QSortFilterProxyModel
+    {
+        public:
+            bool lessThan( const QModelIndex &left, const QModelIndex &right ) const wxOVERRIDE
+            {
+                const QVariant leftData = sourceModel()->data( left );
+                const QVariant rightData = sourceModel()->data( right );
+
+                if ( leftData.type() != QVariant::String )
+                    return false;
+
+                int insensitiveResult = QString::compare(
+                        leftData.value<QString>(),
+                        rightData.value<QString>(),
+                        Qt::CaseInsensitive );
+
+                if ( insensitiveResult == 0 )
```

Comment:
```suggestion
                if ( !insensitiveResult )
```
...but the name might actually be confusing now...